tensorflow/compiler/jit/xla_gpu_device.cc

```diff
@@ -33,8 +33,34 @@ class XlaGpuDeviceFactory : public DeviceFactory {
  public:
   Status CreateDevices(const SessionOptions& options, const string& name_prefix,
                        std::vector<std::unique_ptr<Device>>* devices) override;
+  // Returns a set containing the device ids contained in visible_device_list or
+  // -1 if the string is empty.
+  static xla::StatusOr<std::set<int>> ParseVisibleDeviceList(
+      const string& visible_device_list);
 };
 
+xla::StatusOr<std::set<int>> XlaGpuDeviceFactory::ParseVisibleDeviceList(
+    const string& visible_device_list) {
+  std::set<int> gpu_ids;
+  if (visible_device_list.length() == 0) {
```

Comment:
check `.empty()` instead of `length() == 0`

Reply:
Done.